app/models/utils/bulk_media_packager.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require "sys/filesystem"
+require "zip"
+require "fileutils"
+
+module Utils
+  # Utility to support a bulk image download operation
+  class BulkMediaPackager
+    include ActiveModel::Model
+    # Space we want to leave on disk in mb in base 2
+    DISK_ALLOWANCE = 2048
+    TMP_DIR = "tmp/bulk_images"
+
+    attr_accessor :ability, :search, :operation
+
+    def media_objects_scope
+      responses = Response.accessible_by(@ability, :export)
+      responses = apply_search_scope(responses, @search, @operation.mission) if @search.present?
+      responses.joins(:answers)
+        .joins("INNER JOIN media_objects ON media_objects.answer_id = answers.id")
+        .joins("INNER JOIN active_storage_attachments ON
+          active_storage_attachments.record_id = media_objects.id")
+        .joins("INNER JOIN active_storage_blobs ON
+          active_storage_blobs.id = active_storage_attachments.blob_id")
+    end
+
+    def calculate_media_size
+      media_objects_scope.sum("active_storage_blobs.byte_size")
+    end
+
+    def space_on_disk?
+      stat = Sys::Filesystem.stat("/")
+      # need to leave space for images, zip file, and copy of zip file while attaching to operation
+      space_left = bytes_to_mb(stat.block_size * stat.blocks_available) -
+        bytes_to_mb(calculate_media_size * 2)
+      space_left >= DISK_ALLOWANCE
+    end
+
+    def download_and_zip_images
+      Dir.mkdir(Rails.root.join(TMP_DIR)) unless
+        File.exist?(Rails.root.join(TMP_DIR))
```

Comment:
Tip: `mkdir_p` is safer and allows you to avoid the existence check.

Reply:
good to know! thx